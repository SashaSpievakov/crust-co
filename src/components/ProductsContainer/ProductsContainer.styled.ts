import styled from 'styled-components';

interface SectionProps {
  noSearchResult: boolean;
}

const Section = styled.section<SectionProps>`
  display: grid;
  grid-template-columns: ${({ noSearchResult }) =>
    noSearchResult ? '1fr' : 'repeat(3, 1fr)'};
  gap: 40px 100px;
  justify-items: center;
  align-items: center;

  @media (max-width: 1150px) {
    grid-template-columns: ${({ noSearchResult }) =>
      noSearchResult ? '1fr' : 'repeat(2, 1fr)'};
  }

  @media (max-width: 800px) {
    gap: 40px 50px;
  }

  @media (max-width: 700px) {
    grid-template-columns: ${({ noSearchResult }) =>
      noSearchResult ? '1fr' : 'repeat(1, 1fr)'};
    gap: 0;
  }
`;

const SearchError = styled.h3`
  font-size: 35px;
  margin: 100px 0;

  @media (max-width: 400px) {
    font-size: 30px;
    padding: 0 10px;
  }
`;

export { Section, SearchError };
