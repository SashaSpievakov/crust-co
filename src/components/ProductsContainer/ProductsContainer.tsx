import { useAppSelector } from '../../hooks/reduxHooks';
import { IPizzaItem } from '../../models/IPizzaItem';
import { selectSearchValue } from '../../store/slices/searchSlice';
import ItemCard from '../ItemCard/ItemCard';
import Skeleton from '../ItemCard/Skeleton';
import Container from './ProductsContainer.styled';

interface ProductsContainerProps {
  isLoading: boolean;
  items: IPizzaItem[];
}

const ProductsContainer = ({ isLoading, items }: ProductsContainerProps) => {
  const searchValue = useAppSelector(selectSearchValue);

  return (
    <Container>
      {isLoading
        ? [...new Array(9)].map((_, i) => <Skeleton key={i} />)
        : items
            .filter((item: IPizzaItem) =>
              item.name
                .toLowerCase()
                .includes(searchValue.trim().toLowerCase()),
            )
            .map((item: IPizzaItem) => <ItemCard key={item.id} {...item} />)}
    </Container>
  );
};
export default ProductsContainer;
