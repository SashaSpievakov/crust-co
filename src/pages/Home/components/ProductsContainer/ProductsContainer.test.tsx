import '@testing-library/jest-dom';

import { screen } from '@testing-library/react';

import {
  rendererWithProviders,
  renderWithProvidersAndRoutes,
} from '@src/tests/helpers';
import { mockItems } from '@src/tests/mocks/mockData/mockData';

import { ProductsContainer } from './ProductsContainer';

describe('ProductsContainer tests', () => {
  test('should render the ProductsContainer component', () => {
    const snapshot = rendererWithProviders(
      <ProductsContainer isLoading={false} items={mockItems} searchValue="" />,
    );
    expect(snapshot).toMatchSnapshot();
  });

  test('should render the ProductsContainer component with not matching search value', () => {
    const snapshot = rendererWithProviders(
      <ProductsContainer
        isLoading={false}
        items={mockItems}
        searchValue="testing value"
      />,
    );
    expect(snapshot).toMatchSnapshot();
  });

  test('should render the ProductsContainer component with loading skeletons', () => {
    renderWithProvidersAndRoutes(
      <ProductsContainer isLoading items={mockItems} searchValue="" />,
    );
    const skeletons = screen.getAllByRole('presentation');

    expect(skeletons[0]).toBeInTheDocument();
    expect(skeletons[8]).toBeInTheDocument();
  });

  test('should render items with a search value', () => {
    renderWithProvidersAndRoutes(
      <ProductsContainer
        isLoading={false}
        items={mockItems}
        searchValue="  Shrimp piZza   "
      />,
    );
    const heading = screen.getByRole('heading', { name: /shrimp pizza/i });

    expect(heading).toBeInTheDocument();
  });
});
