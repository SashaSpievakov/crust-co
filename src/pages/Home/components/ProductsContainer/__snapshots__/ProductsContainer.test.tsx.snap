// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsContainer tests should render the ProductsContainer component 1`] = `
Object {
  "children": Array [
    <article
      className="sc-jRwbcX giLUud"
    >
      <a
        className="sc-fvEvSO jSMwtJ"
        href="/pizzas/Shrimp Pizza"
        onClick={[Function]}
      >
        <picture>
          <source
            srcSet="./assets/img/pizzas/pizza3.webp"
            type="image/webp"
          />
          <source
            srcSet="./assets/img/pizzas/pizza3.png"
            type="image/png"
          />
          <img
            alt="Shrimp Pizza pizza"
            className="sc-fcCDlF figEwF"
            height="250"
            src="./assets/img/pizzas/pizza3.png"
            width="250"
          />
        </picture>
        <h3
          className="sc-GKYbw gOnPJi"
        >
          Shrimp Pizza
        </h3>
      </a>
      <div
        className="sc-jNJNQp djLnsE"
      >
        <ul>
          <li
            aria-current={true}
            className="sc-hlLBRy hHFlzz"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            traditional
          </li>
          <li
            aria-current={false}
            className="sc-hlLBRy iHgFco"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            thin
          </li>
        </ul>
        <ul>
          <li
            aria-current={true}
            className="sc-hlLBRy hHFlzz"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            12
             inch
          </li>
          <li
            aria-current={false}
            className="sc-hlLBRy iHgFco"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            14
             inch
          </li>
          <li
            aria-current={false}
            className="sc-hlLBRy iHgFco"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            16
             inch
          </li>
        </ul>
      </div>
      <section
        className="sc-gsGlKL dFNTzs"
      >
        <h3
          className="sc-bWOGAC jKsOjm"
        >
          10
          $
        </h3>
        <button
          className="sc-bcXHqe sc-hLBbgP sc-jSUZER glVyWT bcVaXZ dcTKOB"
          onClick={[Function]}
        >
          <svg
            className="sc-ZqFbI dumUov"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"
            />
          </svg>
          <span>
            Add
          </span>
        </button>
      </section>
    </article>,
    <article
      className="sc-jRwbcX giLUud"
    >
      <a
        className="sc-fvEvSO jSMwtJ"
        href="/pizzas/Pepperoni Pizza"
        onClick={[Function]}
      >
        <picture>
          <source
            srcSet="./assets/img/pizzas/pizza2.webp"
            type="image/webp"
          />
          <source
            srcSet="./assets/img/pizzas/pizza2.png"
            type="image/png"
          />
          <img
            alt="Pepperoni Pizza pizza"
            className="sc-fcCDlF figEwF"
            height="250"
            src="./assets/img/pizzas/pizza2.png"
            width="250"
          />
        </picture>
        <h3
          className="sc-GKYbw gOnPJi"
        >
          Pepperoni Pizza
        </h3>
      </a>
      <div
        className="sc-jNJNQp djLnsE"
      >
        <ul>
          <li
            aria-current={true}
            className="sc-hlLBRy hHFlzz"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            traditional
          </li>
        </ul>
        <ul>
          <li
            aria-current={true}
            className="sc-hlLBRy hHFlzz"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            12
             inch
          </li>
          <li
            aria-current={false}
            className="sc-hlLBRy iHgFco"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            14
             inch
          </li>
        </ul>
      </div>
      <section
        className="sc-gsGlKL dFNTzs"
      >
        <h3
          className="sc-bWOGAC jKsOjm"
        >
          12
          $
        </h3>
        <button
          className="sc-bcXHqe sc-hLBbgP sc-jSUZER glVyWT bcVaXZ dcTKOB"
          onClick={[Function]}
        >
          <svg
            className="sc-ZqFbI dumUov"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"
            />
          </svg>
          <span>
            Add
          </span>
        </button>
      </section>
    </article>,
    <article
      className="sc-jRwbcX giLUud"
    >
      <a
        className="sc-fvEvSO jSMwtJ"
        href="/pizzas/Diablo"
        onClick={[Function]}
      >
        <picture>
          <source
            srcSet="./assets/img/pizzas/pizza6.webp"
            type="image/webp"
          />
          <source
            srcSet="./assets/img/pizzas/pizza6.png"
            type="image/png"
          />
          <img
            alt="Diablo pizza"
            className="sc-fcCDlF figEwF"
            height="250"
            src="./assets/img/pizzas/pizza6.png"
            width="250"
          />
        </picture>
        <h3
          className="sc-GKYbw gOnPJi"
        >
          Diablo
        </h3>
      </a>
      <div
        className="sc-jNJNQp djLnsE"
      >
        <ul>
          <li
            aria-current={true}
            className="sc-hlLBRy hHFlzz"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            traditional
          </li>
          <li
            aria-current={false}
            className="sc-hlLBRy iHgFco"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            thin
          </li>
        </ul>
        <ul>
          <li
            aria-current={true}
            className="sc-hlLBRy hHFlzz"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            12
             inch
          </li>
          <li
            aria-current={false}
            className="sc-hlLBRy iHgFco"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex={0}
          >
            16
             inch
          </li>
        </ul>
      </div>
      <section
        className="sc-gsGlKL dFNTzs"
      >
        <h3
          className="sc-bWOGAC jKsOjm"
        >
          15
          $
        </h3>
        <button
          className="sc-bcXHqe sc-hLBbgP sc-jSUZER glVyWT bcVaXZ dcTKOB"
          onClick={[Function]}
        >
          <svg
            className="sc-ZqFbI dumUov"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 0a1 1 0 0 1 1 1v6h6a1 1 0 1 1 0 2H9v6a1 1 0 1 1-2 0V9H1a1 1 0 0 1 0-2h6V1a1 1 0 0 1 1-1z"
            />
          </svg>
          <span>
            Add
          </span>
        </button>
      </section>
    </article>,
  ],
  "props": Object {
    "className": "sc-eGugkK eHCMkP",
  },
  "type": "section",
}
`;

exports[`ProductsContainer tests should render the ProductsContainer component with not matching search value 1`] = `
Object {
  "children": Array [
    <h3
      className="sc-csDkEv kRgCHa"
    >
      No pizzas with that name were found...
    </h3>,
  ],
  "props": Object {
    "className": "sc-eGugkK jSTRGl",
  },
  "type": "section",
}
`;
